### ISSUE TITLE
Clarification of variable reassignment vs. object modification may be needed (Variables lesson, Ruby Programming course)

### ISSUE THREAD

**Issue by nguyenjessev** [TheOdinProject/curriculum#19545]
In the "Variables are References" section of the Variables lesson in the Ruby Programming course, an example is given of a variable being assigned to a string and a second variable assigned to the first variable.

A bang method is used to modify the object that the first variable is pointing to, and then it is shown that the second variable's value also changes.

However, at this point in the course bang methods have not been introduced yet. It is hinted at that the bang method is modifying the string that is stored at the memory location that both variables are pointing to, but this could be a difficult concept to grasp for a beginning learner, and they may get the impression that variable reassignment would have the same effect. Perhaps some clarification is needed to explain this particular situation?

![image](https://user-images.githubusercontent.com/67484662/88482075-00ac6f00-cf2d-11ea-87cb-fa0ebd186dcb.png)

**Comment by Drew-Goddyn** [comment]
@nguyenjessev thanks for the suggestion. Sometimes its hard to explain concepts when you're familiar with them for so long. Do you have any suggestions or maybe an example of how it can be rewritten?

Feel free to open a pull request with ideas you have. Even if they aren't merged it it's still a good way to document feedback!

**Comment by dm-murphy** [comment]
@nguyenjessev @Drew-Goddyn 

This section tripped me up the first time I went through it. So I looked it up and read more on bang methods and also think one of the article resources on the page makes mention of bangs too. But I took away the wrong lesson by focusing on the bang itself. I failed to understand that the variable, `johns_location`, was pointing to the same object as `desired_location` and that by modifying the object through the `johns_location` variable (bang or no bang) I was modifying the same object `desired_location` points to. 

I'm guessing I'm not the only beginner that could be confused here. But I also think this could be one of those things where playing with it in IRB is better than just being told what's going on. 

I'd suggest that after the second code block, the curriculum should ask the student if they can figure out what happened here and to play around with setting these variables and modifying them in IRB. 

I can submit this as a PR when I get the chance if no one else jumps on it.
